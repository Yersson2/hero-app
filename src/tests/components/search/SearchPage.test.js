import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter, Route } from 'react-router';
import { SearchPage } from '../../../components/search/SearchPage';

describe('test in <Search />', () => {
  
  test('should show correctly with default values ', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries = {['/search']}>
        <Route path = "/search" component = { SearchPage } />
      </MemoryRouter>
    );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('.alert-info').text().trim() ).toBe('Search a hero');
    
  });

  test('should show batman and the input with the queryString value', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries = {['/search?q=batman']}>
        <Route path = "/search" component = { SearchPage } />
      </MemoryRouter>
    );

    expect( wrapper.find('input').prop('value') ).toBe('batman');
    expect( wrapper ).toMatchSnapshot();
    
  });

  test('should show an error if does not find the hero', () => {

    const wrapper = mount(
      <MemoryRouter initialEntries = {['/search?q=batman123']}>
        <Route path = "/search" component = { SearchPage } />
      </MemoryRouter>
    );

    expect( wrapper.find('.alert-danger').text().trim() ).toBe(`There isn't a hero with batman123`);
    expect( wrapper ).toMatchSnapshot();
    
  });

  test("should call the history's push", () => {

    const history = {
      push: jest.fn()
    };

    const wrapper = mount(
      <MemoryRouter initialEntries = {['/search?q=batman123']}>
        <Route 
          path = "/search" 
          component = { () => <SearchPage history = { history } /> } />
      </MemoryRouter>
    );

    wrapper.find('input').simulate( 'change', {
      target: {
        name: 'searchText',
        value: 'batman'
      }
    });

    wrapper.find('form').prop('onSubmit')({
      preventDefault(){}
    });

    expect( history.push ).toHaveBeenCalledWith(`?q=batman`);
    
  })
  
  
  
  
})
