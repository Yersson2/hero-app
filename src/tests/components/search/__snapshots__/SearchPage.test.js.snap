// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <Search /> should show an error if does not find the hero 1`] = `
<div>
  <h1>
    Search Page
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman123"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There isn't a hero with 
        batman123
      </div>
    </div>
  </div>
</div>
`;

exports[`test in <Search /> should show batman and the input with the queryString value 1`] = `
<div>
  <h1>
    Search Page
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                Batman
              </h5>
              <p
                className="card-text"
              >
                Bruce Wayne
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                MÃ¡s...
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test in <Search /> should show correctly with default values  1`] = `
<div>
  <h1>
    Search Page
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
